lib/vm.js

```diff
@@ -286,6 +287,70 @@ function runInThisContext(code, options) {
   return createScript(code, options).runInThisContext(options);
 }
 
+function compileFunction(code, params, options) {
+  if (typeof code !== 'string') {
+    throw new ERR_INVALID_ARG_TYPE('code', 'string', code);
+  }
+  if (params !== undefined) {
+    if (!Array.isArray(params)) {
+      throw new ERR_INVALID_ARG_TYPE('params', 'Array', params);
+    }
+    params.forEach((param, i) => {
+      if (typeof param !== 'string') {
+        throw new ERR_INVALID_ARG_TYPE(`params[${i}]`, 'string', param);
+      }
+    });
+  }
+
```

Comment:
We've generally tried to avoid using `forEach` within core functions because it's an order of magnitude slower than a regular for-loop. Even if this is not expected to be a hot function, I think I'd rather see this as a regular for-loop.

Reply:
if we're taking the performance approach, `forEach` has actually become faster in the common case than a for loop (in V8 anyway).

Reply:
👆 \cc @bmeurer

Reply:
Whatever the outcome of this discussion, please let’s not hold up the PR over this – this isn’t hot code (or at least the cost of actual parsing + compilation is likely going to overshadow any choice of this kind), so I think it’s fine to go with whatever the PR author considers most readable.